apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/LoggingSnippets.cs

```diff
@@ -0,0 +1,180 @@
+        }
+
+        /// <summary>
+        /// Test for RegisterGoogleLogger, RegisterGoogleLogger2 and UseGoogleLogger snippets.
+        /// The RegisterGoogleLogger and RegisterGoogleLogger2 snippets are configuration snippets
+        /// so the best way to test it is by checking if logging works properly.
+        /// </summary>
+        [Fact]
+        public async Task Logs_Information()
+        {
+            await _client.GetAsync($"/LoggingSamples/LogInformation/{_testId}");
+
+            PollAndVerifyLog(_startTime, _testId);
+        }
+
+        internal static void PollAndVerifyLog(DateTime startTime, string testId)
+        {
+            var results = s_polling.GetEntries(startTime, testId, 1, LogSeverity.Info);
+            results = from result in results
+                      where result.JsonPayload?.Fields["log_name"]?.StringValue?.Equals(LoggingSamplesController.LoggerName) ?? false
+                      select result;
+
+            Assert.Equal(1, results.Count());
+
+            var logEntry = results.FirstOrDefault();
+            Assert.Equal(LogSeverity.Info, logEntry.Severity);
+            Assert.True(logEntry.JsonPayload?.Fields["message"]?.StringValue?.Contains(testId));
+        }
+
+        public void Dispose()
+        {
+            _client.Dispose();
+            _server.Dispose();
+        }
+    }
+
+    /// <summary>
+    /// A simple web application to test the <see cref="GoogleLogger"/>
+    /// and associated classes.
+    /// </summary>
+    internal class LoggingTestApplication
+    {
+        private static readonly string ProjectId = Utils.GetProjectIdFromEnvironment();
+
+        // Sample: RegisterGoogleLogger
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+        }
+
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
+        {
+            // Replace ProjectId with your Google Cloud Project ID.
+            loggerFactory.AddGoogle(app.ApplicationServices, ProjectId);
+
+            // ...any other configuration your application requires.
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+        // End sample
+    }
+
+    /// <summary>
+    /// A simple web application to test the <see cref="GoogleLogger"/>
+    /// and associated classes in ASP.Net Core 2.
+    /// </summary>
+    internal class LoggingTestApplicationCore2
+    {
+        private static readonly string ProjectId = Utils.GetProjectIdFromEnvironment();
+
+        // Sample: RegisterGoogleLogger2
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Replace ProjectId with your Google Cloud Project ID.
+            services.AddSingleton<ILoggerProvider>(sp => GoogleLoggerProvider.Create(sp, ProjectId));
```

Comment:
The suggested way for 2.0 is to do this in the webbulder like this: http://googlecloudplatform.github.io/google-cloud-dotnet/docs/Google.Cloud.Diagnostics.AspNetCore/#initializing-logging-aspnet-core-20

Reply:
Yes, I got caught up in the Startup thing for all services. Done